symengine/functions.cpp

```diff
@@ -610,8 +610,13 @@ RCP<const Basic> floor(const RCP<const Basic> &arg)
         RCP<const Number> s = down_cast<const Add &>(*arg).get_coef();
         umap_basic_num d = down_cast<const Add &>(*arg).get_dict();
         if (is_a<Integer>(*s)) {
-            return add(
-                s, make_rcp<const Floor>(Add::from_dict(zero, std::move(d))));
+            if (d.size() == 1) {
+                return add(
+                    s, floor(Add::from_dict(zero, std::move(d))));
```

Comment:
Why not return this for both cases? (To avoid recursion, we have to check if `s==0`)